cardano-cli/src/Cardano/CLI/EraBased/Commands/Transaction.hs

```diff
@@ -133,6 +133,8 @@ data TransactionBuildCmdArgs era = TransactionBuildCmdArgs
   , mUpdateProposalFile :: !(Maybe (Featured ShelleyToBabbageEra era (Maybe UpdateProposalFile)))
   , voteFiles :: ![(VoteFile In, Maybe CliVoteScriptRequirements)]
   , proposalFiles :: ![(ProposalFile In, Maybe CliProposalScriptRequirements)]
+  , disableMetadataChecksOnProposals :: !Bool
```

Comment:
Don't use `Bool`s. It's not too bad here but boolean blindness is a thing: https://yveskalume.dev/boolean-blindness-dont-represent-state-with-boolean

Reply:
In particular for a check flag it can be confusing, having a type like:

```hs
data MetadataCheck =
  CheckMetadata
  | DontCheckMetadata
```

will be clearer